acsys/status.py

```diff
@@ -2,52 +2,89 @@
         ACNET_NOREMMEM = 1 + 256 * -3
+        """no remote memory available"""
         ACNET_RPLYPACK = 1 + 256 * -4
+        """reply message packet assembly error"""
         ACNET_REQPACK = 1 + 256 * -5
+        """request message packet assembly error"""
         ACNET_REQTMO = 1 + 256 * -6
+        """request timeout with queued at destination"""
         ACNET_QUEFULL = 1 + 256 * -7
+        """request failed, destination queue full"""
         ACNET_BUSY = 1 + 256 * -8
+        """request failed, destination task busy"""
         ACNET_NOT_CONNECTED = 1 + 256 * -21
+        """not connected to the network"""
         ACNET_ARG = 1 + 256 * -22
+        """missing arguments"""
         ACNET_IVM = 1 + 256 * -23
+        """invalid message length or buffer address"""
         ACNET_NO_SUCH = 1 + 256 * -24
+        """no such request or reply"""
         ACNET_REQREJ = 1 + 256 * -25
+        """request to destination task was rejected"""
         ACNET_CANCELLED = 1 + 256 * -26
+        """request has been cancelled"""
         ACNET_NAME_IN_USE = 1 + 256 * -27
+        """task name already in use"""
         ACNET_NCR = 1 + 256 * -28
+        """not connected as a RUM task"""
         ACNET_NO_NODE = 1 + 256 * -30
+        """no such logical node"""
         ACNET_TRUNC_REQUEST = 1 + 256 * -31
+        """truncated request"""
         ACNET_TRUNC_REPLY = 1 + 256 * -32
+        """truncated reply"""
         ACNET_NO_TASK = 1 + 256 * -33
+        """no such destination task"""
         ACNET_DISCONNECTED = 1 + 256 * -34
+        """replier task being disconnected"""
         ACNET_LEVEL2 = 1 + 256 * -35
+        """ACNET level II function error"""
         ACNET_HARD_IO = 1 + 256 * -41
+        """hard I/O error"""
         ACNET_NODE_DOWN = 1 + 256 * -42
+        """logical node down or offline"""
         ACNET_SYS = 1 + 256 * -43
+        """system service error"""
         ACNET_NXE = 1 + 256 * -44
+        """untranslatable error"""
         ACNET_BUG = 1 + 256 * -45
+        """network internal error"""
         ACNET_NE1 = 1 + 256 * -46
+        """VMS exceeded some quota or limit"""
         ACNET_NE2 = 1 + 256 * -47
+        """VMS no address for request/reply ID word"""
         ACNET_NE3 = 1 + 256 * -48
+        """VMS buffer/control block vector in use or block already locked"""
         ACNET_UTIME = 1 + 256 * -49
+        """user-generated network timeout"""
         ACNET_INVARG = 1 + 256 * -50
+        """invalid argument passed"""
         ACNET_MEMFAIL = 1 + 256 * -51
+        """memory allocation failed"""
         ACNET_NO_HANDLE = 1 + 256 * -52
+        """requested handle doesn't exist"""
 
     def __init__(self, val):
         """Creates a status value which is initialized with the supplied
         value. The value must be in the range of signed, 16-bit
         integers.
-        Do not use this method directly, use Status.create(val) instead.
+        Do not use this method directly, use Status.create(val), or
```

Comment:
You could change `Codes` to `_Codes` which is the convention for signaling a private property.

Reply:
Why?

Reply:
Because you are saying in words here can be signaled, a bit programmatically, through a common Python convention.

Reply:
I'm not sure I understand, changing `Codes` to `_Codes` doesn't tell people that they shouldn't do `st = Status(0xFF)`, does it?

Reply:
I reread this more closely. I thought the `__init__` was under `Codes` but it's under `Status`, my mistake.
There isn't a concept of private constructor concept in Python. The underscore is a common trust that people won't use it 😆 
You can see from this documentation that Python does not import things with the leading underscore and I think it's been adopted further. https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles
Maybe this is an option? https://stackoverflow.com/questions/8212053/private-constructor-in-python
Or maybe a better explanation, https://stackify.dev/482763-private-constructor-in-python.

Reply:
Yeah, I'd read those also, the stackify page is just a reformatting of stackoverflow's thread. I'll go with the `__key` solution then.

Reply:
arrrgggg.... if I do the `__create_key` solution, then I cannot call the `Status` constructor from the child classes, maybe I should flag it as an Abstract class.